cmd/root.go

```diff
@@ -50,6 +50,7 @@ Find more information at https://github.com/kubeapps/installer.`,
 func init() {
 	RootCmd.PersistentFlags().CountP("verbose", "v", "Increase verbosity.")
 	RootCmd.PersistentFlags().Set("logtostderr", "true")
+	RootCmd.PersistentFlags().String("cloud-provider", "gke", "Specify the cloud provider.")
```

Comment:
in my opinion:
 - This flag should be a choice selection (the allowed values would be `gke` and `minikube`
 - The default value should be empty
 - If there is not auto detection this flag should be mandatory

Reply:
I think `--cloud_provider` really means k8s running on cloud. I haven't had chance to test the installer on aws or other _popular_ clouds. I am not sure if we have to do any special action on aws like we do on gke (granting that special permission), so we support the flag there for further _switch cases_. Minikube should not be listed on this flag (or if it does, we should call `--cloud-provider=none`)

Reply:
yeah, I was thinking in changing the flag name or `--cloud-provider=none` seems fine as well. In any case I think it should be a choice parameter (`--cloud-provider=google` would do nothing but it will be allowed).

BTW, what happens if in minikube you try to add the cluster role binding?

Reply:
Interesting... On minikube it's still being created without issues. But we won't want to do that. So choices:

1) `--cloud-provider` default to `none` then on minikube just run `kubeapps up`, on gke `kubeapps up --cloud-provider=gke`
2)`--cloud-provider` default to `gke` then on gke just run `kubeapps up`, on minikube run `kubeapps up --cloud-provider=none` 
3) default to empty. I don't think we should go with this -> longer typing for user.

Reply:
For this case I think we should go for the longer typing. If I am in GKE and I forgot to use the `--cloud-provider=gke` I would need to figure it out why the deployment is failing. Or defaulting to`gke` since using the CRB in minikube is not critical?

Reply:
Forget about that flag for now. I would do a simple detection based on the k8s version convention on gke: 

```
$ gcloud container get-server-config
...
validMasterVersions:
- 1.8.3-gke.0
- 1.8.2-gke.0
- 1.8.1-gke.1
- 1.7.10-gke.0
- 1.7.9-gke.0
- 1.7.8-gke.0
- 1.6.11-gke.0
```

If there is something changes in the way gke labeling k8s version in future, we can always update the installer.